src/components/locations/components/NewLocation.js

```diff
@@ -1,79 +1,92 @@
-import React, { useState } from 'react';
+import React from 'react';
 import { Form, Input, Icon, Button, Card, Upload, Row, Col } from 'antd';
 import { useSelector, useDispatch } from 'react-redux';
-import { CREATE_LOCATION, CREATE_NEW_LOCATION, FORM_ITEMS, UPLOAD_TEXT } from '../constants';
+import {
+    CREATE_LOCATION, CREATE_NEW_LOCATION, FORM_ITEMS,
+    UPLOAD_TEXT, FORM_ITEM_LAYOUT, WRAPPER_COL
+} from '../constants';
+import { newLocations } from '../actions';
+import { normFile } from '../../utils/functions';
 
 const NewLocation = ({ form }) => {
-    // const [state, setstate] = useState(initialState);
-    // const dispatch = useDispatch();
+    const dispatch = useDispatch();
     const { loading } = useSelector(state => state.signIn);
     const { getFieldDecorator, resetFields, validateFields } = form;
+    const handleFormData = formValues => {
+        const formData = new FormData();
+        Object.keys(formValues).forEach(key => {
+            formData.append(key, formValues[key]);
+            if (key === 'images') {
+                formValues[key].forEach(element => {
+                    formData.append('images', element.originFileObj);
+                });
+            }
+        });
+        return formData;
+    };
     const handleSubmit = e => {
         e.preventDefault();
-        validateFields((err, values) => {
+        validateFields(async (err, values) => {
             if (!err) {
-                const formValues = {
-                    email: values.email,
-                    password: values.password,
-                };
-                // dispatch(userSignIn(formValues));
-                resetFields();
+                const newFormData = handleFormData(values);
+                dispatch(newLocations(newFormData));
+                setTimeout(() => {
+                    resetFields();
+                }, 3000);
```

Comment:
what purpose those this setTimeout serve?

Reply:
Absoutely not needed. Was testing out something and forgot to remove it at the end.